frontend/apps/reader/modules/readerhandmade.lua

```diff
@@ -532,12 +538,28 @@ function ReaderHandMade:addOrEditPageTocItem(pageno, when_updated_callback, sele
                         text = _("Use selected text"),
                         callback = function()
                             -- Just replace the text without saving, to allow editing/fixing it
-                            dialog:setInputText(selected_text.text, nil, false)
+                            dialog:setInputText(selected_text.text, nil, true)
                         end,
                     } or nil,
             } or nil,
         },
     }
+    if no_dialog then
+        if item_found then return true end  -- no changes if existing TOC entry
+        if selected_text then -- via highlight dialog
```

Comment:
You may as well move that above when we start creating the dialog.
No need to create it if we end up not using it :)